File: pset1/cash/cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>

int main(void)
{

    float change;

    // Prompt user for amount of change owed until positive float provided
    do
    {
        printf("How much change is owed?\n");
        change = get_float();
    }
    while (change < 0);

    // Convert float to integer
    int change_int = (int)round(change * 100);
    int coins = 0;

    // Calculate number of quarters in change
    while (change_int >= 25)
    {
        change_int -= 25;
        coins++;
    }

    // Calculate number of dimes in change
    while (change_int >= 10)
    {
        change_int -= 10;
        coins++;
    }

    // Calculate number of nickels in change
    while (change_int >= 5)
    {
        change_int -= 5;
        coins++;
    }

    // Calculate number of pennys in change
    while (change_int >= 1)
    {
        change_int -= 1;
        coins++;
    }

    // Print minimum number of coins required in change
    printf("It turns out you need %i coins in total\n", coins);
}

File: pset6/cash/cash.py
from cs50 import get_int, get_float

# Initialize change + coins variables
change = -1.00
coins = 0

# Prompt user for change value
while change <= 0:
    change = get_float("how much change is owed?\n")

# Convert change to int
change = round(change * 100)
print("{}" .format(change))

# Calculate number of 25c coins
while change >= 25:
    change -= 25
    coins += 1

# Calculate number of 10c coins
while change >= 10:
    change -= 10
    coins += 1

# Calculate number of 5c coins
while change >= 5:
    change -= 5
    coins += 1

# Calculate number of 1c coins
while change >= 1:
    change -= 1
    coins += 1

print("{}" .format(coins))